relays/bin-substrate/src/cli/swap_tokens.rs

```diff
@@ -0,0 +1,642 @@
+					TO_MILLAU_ESTIMATE_MESSAGE_FEE_METHOD as ESTIMATE_TARGET_TO_SOURCE_MESSAGE_FEE_METHOD,
+					WITH_RIALTO_TOKEN_SWAP_PALLET_NAME as TOKEN_SWAP_PALLET_NAME,
+				};
+				use bp_rialto::{
+					derive_account_from_millau_id as derive_target_account_from_source_account,
+					TO_RIALTO_ESTIMATE_MESSAGE_FEE_METHOD as ESTIMATE_SOURCE_TO_TARGET_MESSAGE_FEE_METHOD,
+				};
+
+				const SOURCE_CHAIN_ID: bp_runtime::ChainId = bp_runtime::MILLAU_CHAIN_ID;
+				const TARGET_CHAIN_ID: bp_runtime::ChainId = bp_runtime::RIALTO_CHAIN_ID;
+
+				const SOURCE_SPEC_VERSION: u32 = millau_runtime::VERSION.spec_version;
+				const TARGET_SPEC_VERSION: u32 = rialto_runtime::VERSION.spec_version;
+
+				const SOURCE_TO_TARGET_LANE_ID: bp_messages::LaneId = *b"swap";
+				const TARGET_TO_SOURCE_LANE_ID: bp_messages::LaneId = [0, 0, 0, 0];
+
+				$generic
+			}
+		}
+	};
+}
+
+impl SwapTokens {
+	/// Run the command.
+	pub async fn run(self) -> anyhow::Result<()> {
+		select_bridge!(self.bridge, {
+			let source_client = self.source.to_client::<Source>().await?;
+			let source_sign = self.source_sign.to_keypair::<Target>()?;
+			let target_client = self.target.to_client::<Target>().await?;
+			let target_sign = self.target_sign.to_keypair::<Target>()?;
+
+			// names of variables in this function are matching names used by the `pallet-bridge-token-swap`
+
+			// prepare token swap intention
+			let token_swap = self
+				.prepare_token_swap::<Source, Target>(&source_client, &source_sign, &target_sign)
+				.await?;
+
+			// group all accounts that will be used later
+			let accounts = TokenSwapAccounts {
+				source_account_at_bridged_chain: derive_target_account_from_source_account(
+					bp_runtime::SourceAccount::Account(token_swap.source_account_at_this_chain.clone()),
+				),
+				target_account_at_this_chain: derive_source_account_from_target_account(
+					bp_runtime::SourceAccount::Account(token_swap.target_account_at_bridged_chain.clone()),
+				),
+				source_account_at_this_chain: token_swap.source_account_at_this_chain.clone(),
+				target_account_at_bridged_chain: token_swap.target_account_at_bridged_chain.clone(),
+				swap_account: FromSwapToThisAccountIdConverter::convert(token_swap.using_encoded(blake2_256).into()),
+			};
+
+			// account balances are used to demonstrate what's happening :)
+			let initial_balances = read_account_balances(&accounts, &source_client, &target_client).await?;
+
+			// before calling something that may fail, log what we're trying to do
+			log::info!(target: "bridge", "Starting swap: {:?}", token_swap);
+			log::info!(target: "bridge", "Swap accounts: {:?}", accounts);
+			log::info!(target: "bridge", "Initial account balances: {:?}", initial_balances);
+
+			//
+			// Step 1: swap is created
+			//
+
+			// prepare `Currency::transfer` call that will happen at the target chain
+			let bridged_currency_transfer: CallOf<Target> = pallet_balances::Call::transfer(
```

Comment:
There's a lot of code in this function - I've been trying to split it into several methods. But there's a problem - pallet calls are generic over runtime. So I need to pass runtime as generic arguments to these methods. But if this'll be extended to something behind Millau<>Rialto, then we won't have an access to the runtime => I've decided not to split it further.

I could also add call-craft functions to the `select_bridge!()` macro if that's unacceptable.

Reply:
Let's leave it like that for now, I think it will soon be replaced by XCM anyway, so this should be just an example for now.